### ISSUE TITLE
ENH: Would you consider making q available through conda?

### ISSUE THREAD

**Issue by Sann5** [harelba/q#322]
I would like to use q in a cluster environment. However, the cluster does not support brew but it supports conda (there are ways of making q work but it's a dirty solution). I believe the process of building a package and making it available in conda is relatively uncomplicated. Would you consider doing so? I could also do it and link this repo for credits if that is something you would be okay with. 

Thanks for the great tool! Cheers!

**Comment by harelba** [comment]
Hi @Sann5 sorry for missing this message, just saw it now, crazy times.

Adding q to conda would be great. I'm not sure about the specifics though, so I'd really appreciate it if you can do it and link this repo and the q site if possible.

Would be great if you can send me the PR that you create for doing it, so i can understand a bit about the process and the concept.

Harel

**Comment by Sann5** [comment]
@harelba should I sign you up as one of the maintainers of the conda-forge recipe (a GitHub repo that will be in charge of publishing q in conda)? Or someone else? There will be SPAM from conda-forge. I will also sign up as a maintainer for the time being.

**Comment by harelba** [comment]
No need to sign me up as a maintainer for now. Once you get tired from being a maintainer, please ping me and I'll take over as needed.

Can you please send me the relevant PR for conda-forge, so i can take a quick look at it?

**Comment by Sann5** [comment]
For sure. I'll ping you when the PR is pending for review.

**Comment by Sann5** [comment]
Hey @harelba. Sorry to bother you with this, but Im having trouble making the recipe to install q with conda. In essence what i'm trying to do (from a Mac) is the following... 

```bash
wget https://github.com/harelba/q/archive/v3.1.6.tar.gz
gunzip q-3.1.6.gz
cd q-3.1.6
conda create --name q python=3.8
conda activate q
PYTHON -m pip install . -vv 
q -h
```

The last command gives an an error:
```bash
Traceback (most recent call last):
  File "/path/to/my/envs/q/bin/q", line 5, in <module>
    from bin.q import run_standalone
ModuleNotFoundError: No module named 'bin'
```

`/path/to/my/envs/q/bin/q` is a script that is generated by the `PYTHON -m pip install . -vv` using the `setup.py`. It looks like this...
```python
#!/Users/santiago/mambaforge/envs/q/bin/PYTHON
# -*- coding: utf-8 -*-
import re
import sys
from bin.q import run_standalone
if __name__ == '__main__':
  sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
  sys.exit(run_standalone())
```

Im not sure why this is not working. I know there are other ways of using and installing `q`but I thought this would be the most straightforward for all platforms. I would really appreciate some input. Cheers!

**Comment by Sann5** [comment]
Ok I found the issue. After running `pip install .` a package should be created in `/path/to/your/envs/name_of_your_env/lib/python3.8/site-packages/`. That does not happen because of how `setup.py` is defined. Essentially one must change the following for it to work. 

```python
...

setup(
...
    package_dir={"": "."},
    packages=setuptools.find_packages(),
...
```

I will make an separate issue for this. Im still exploring how the packaging can be done best (the workaround above installed three packages: `q`, `bin` and `test`. I will try such that only one package is installed).

**Comment by Sann5** [comment]
🔧 ⚠️ Blocked by #327 
@harelba I require merging of the above PR and bumping the version before I continue with the conda recipe.

**Comment by harelba** [comment]
@Sann5 Sorry for the delayed responses, too many things in parallel, and times are a bit crazy. 

My knowledge regarding python packaging structure is limited, I really appreciate your effort on this.
